### ISSUE TITLE
Prepare extracellular potential demo for LFPy (Python)

### ISSUE THREAD

**Issue by halfflat** [arbor-sim/arbor#1036]
Arbor may be able to be well utilized as a back-end for [LFPy](https://github.com/LFPy/LFPy). As a first step, we will make a simple demo in Python that computes the potential at a fixed set of electrode sites from a single morphologically detailed neuron.

Model NEURON python code: [pure_neuron_example.py.txt](https://github.com/arbor-sim/arbor/files/4629768/pure_neuron_example.py.txt)

Once #1034 is merged, we'll need to:
* expose trans membrane current probes and sample data in Python interface, together with metadata;
* use the single-cell Python API to model and run a neuron such as in the example code;
* and in anticipation of running multiple neurons, we will need to implement exact sample timing.

**Comment by halfflat** [comment]
We may also need to enrich the morphology interface (in Python and/or C++) for facilitating geometry queries.

**Comment by halfflat** [comment]
Exact sampling provided by #1046

**Comment by halfflat** [comment]
I will split this into two tasks: C++ version and a Python version. The latter will require a Python interface for the cell-wide trans membrane current probes.

**Comment by halfflat** [comment]
The other piece of Python infrastructure required to produce a Python LFP demo with LFPykit is #1225, which is pending review.

**Comment by halfflat** [comment]
Last remaining component is #1262 for correct transmembrane accounting. Turns out #1225 wasn't the last prerequisite after all!
Task then will be to convert existing LFP demo to an LFPkit-based implementation.

**Comment by espenhgn** [comment]
I've made a basic LFPykit + .swc file example available here: https://github.com/LFPy/LFPykit/blob/master/examples/Example_Arbor_swc.ipynb (via https://github.com/LFPy/LFPykit/pull/129). 

It's not particularly elegant though.

**Comment by halfflat** [comment]
Thank you! I'll take a look!

Cheers,
Sam

**Comment by halfflat** [comment]
> I've made a basic LFPykit + .swc file example available here: https://github.com/LFPy/LFPykit/blob/master/examples/Example_Arbor_swc.ipynb (via [LFPy/LFPykit#129](https://github.com/LFPy/LFPykit/pull/129)).

Thanks, I'll take a look!

**Comment by brenthuisman** [comment]
@espenhgn Would you mind if we include the example in the notebook in the Arbor-repo? Probably we would make some small modifications.

**Comment by espenhgn** [comment]
> @espenhgn Would you mind if we include the example in the notebook in the Arbor-repo? Probably we would make some small modifications.

No objections :) 
I'll probably seize the opportunity to patch any improvements into the original example notebook then.

**Comment by brenthuisman** [comment]
OK, thanks!

**Comment by espenhgn** [comment]
Shall I add the example in a PR to Arbor perhaps?

**Comment by brenthuisman** [comment]
By all means :) Although we've though of having notebooks in the repo, so far we have stuck to good old Python scripts, which also also straightforward to hook up to our testing. (All examples are run and checked for failure, which is not the most comprehensive testing procedure but it's a start.) So, if you want to post the notebook, we'd need to add something to test notebooks (`nb-convert` I think) to our pipeline. I can also take care of the scriptification in the PR if that's too much fiddling for you.

**Comment by espenhgn** [comment]
I can of course just export the existing notebook as a regular Python file as is (and clean it up a bit) and add that to the PR for now.

**Comment by brenthuisman** [comment]
Let's do that, we can think about how to integrate notebooks in our testing at a later point (I assume you'll keep this example as a notebook around)

**Comment by espenhgn** [comment]
Yes, the existing LFPykit examples will remain notebooks. I like Github's ability to render them.

**Comment by espenhgn** [comment]
Ok, I've open the PR. I did not check it with the current master as I encountered some issue with the cmake configuration.
